apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -184,3 +256,87 @@ func validateImageURL(imageURL string) error {
 
 	return nil
 }
+
+func checkStatusAnnotation(bmhstatus *BareMetalHostStatus) error {
+
+	operatstatus_allowed := []string{"", "OK", "discovered", "error", "delayed", "detached"}
+	errortype_allowed := []string{"", "provisioned registration error", "registration error", "inspection error", "preparation error", "provisioning error", "power management error"}
```

Comment:
Could we not hardcode the values here? It will be very trivial to forget to update this place.

Reply:
You mean hardcode the values in the baremetalhost_types file ?

Reply:
@dtantsur should we use the go-validator to validate for values instead ?

Reply:
My problem is that if we add a new value, we'll probably forgot to update both this list and the list in the API types.

Reply:
Right now, I see two ways to resolve this, either use go-validator (in which case we only need to update the baremetalhost_types.go file) or add comments at both places so that both sides are updated.

Reply:
Is there a way to validate the status annotation against the OpenAPI definition of the Status subresource? That would be better than having to duplicate that validation by hand (in a way that can potentially get out of sync). The tricky part will be where to get that OpenAPI definition from - presumably we'd have to retrieve the CRD.